staging/src/k8s.io/component-base/logs/json/json.go

```diff
@@ -88,6 +77,7 @@ func (l *zapLogger) Info(msg string, keysAndVals ...interface{}) {
 		Time:    timeNow(),
 		Message: msg,
 	}
+	trySettingCaller(&entry, 2)
```

Comment:
For new klog Depth functions to work, zaplogger you need to implement logr.DepthLogger interface.